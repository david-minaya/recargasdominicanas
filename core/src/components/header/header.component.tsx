import React from 'react';
import { Text } from '../text/text.component';
import { Icon } from '../icon/icon.component';
import { mergeStyle, Style } from './header.module.css';

interface Props {
  logo?: string;
  title: string;
  name: string;
  style?: Style;
  onOpenProfile?: () => void;
}

export function Header(props: Props) {

  const {
    logo,
    title,
    name,
    style: customStyle,
    onOpenProfile
  } = props;

  const style = mergeStyle(customStyle);

  return (
    <div className={style.container}>
      <img className={style.logo} src={logo}/>
      <Text
        text={title}
        className={style.title}/>
      <div 
        className={style.user}
        onClick={onOpenProfile}>
        <Text className={style.name} text={name}/>
        <Icon className={style.icon} icon='expand_more'/>
      </div>
    </div>
  );
}
