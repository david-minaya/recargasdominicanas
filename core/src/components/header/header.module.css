.container {
  height: 44px;
  display: flex;
  align-items: center;
  background-color: white;
  border-bottom: 1px solid var(--border-color);
  padding: 0px 20px;
}

.logo {
  width: 24px;
  height: 24px;
  margin-right: 20px;
}

.title {
  font-size: 16px;
  font-weight: 500;
}

.user {
  display: flex;
  align-items: center;
  cursor: pointer;
  margin-left: auto;
}

.name {
  font-size: 14px;
  font-weight: 500;
  margin-right: 12px;
}

.icon {
  --size: 18px;
}
