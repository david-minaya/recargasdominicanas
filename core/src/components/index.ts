export * from './icon/icon.component';
export * from './option-icon/option-icon.component';
export * from './text/text.component';
export * from './image/image.component';
export * from './title/title.component';
export * from './outline-card/outline-card.component';
export * from './field-error/field-error.component';
export * from './text-field/text-field.component';
export * from './text-area/text-area.component';
export * from './select-option/select-option.component';
export * from './select/select.component';
export * from './autocomplete/autocomplete.component';
export * from './button/button.component';
export * from './fill-button/fill-button.component';
export * from './outline-button/outline-button.component';
export * from './error/error.component';
export * from './outline-item/outline-item.component';
export * from './detail-item/detail-item.component';
export * from './setting-item/setting-item.component';
export * from './transaction-item/transaction-item.component';
export * from './product-item/product-item.component';
export * from './option/option.component';
export * from './sales-report/sales-report.component';
export * from './brand/brand.component';
export * from './toolbar-title/toolbar-title.component';
export * from './toolbar/toolbar.component';
export * from './header/header.component';
export * from './navbar/navbar.component';
export * from './navbar-option/navbar-option.component';
export * from './menu/menu.component';
export * from './tab/tab.component';
export * from './tab-bar/tab-bar.component';
export * from './modal/modal.component';
export * from './modal-toolbar/modal-toolbar.component';
export * from './modal-content/modal-content.component';
export * from './modal-options/modal-options.component';
export * from './modal-actions/modal-actions.component';
export * from './base-modal/base-modal.component';
export * from './loading-modal/loading-modal.component';
export * from './receipt/receipt.component';
export * from './receipt-modal/receipt-modal.component';
export * from './switch/switch.component';
export * from './switch-case/switch-case.component';
export * from './cancel-transaction-modal/cancel-transaction-modal.component';
export * from './base-cell/base-cell.component';
export * from './cell/cell.component';
export * from './cell-button/cell-button.component';
export * from './cell-icon/cell-icon.component';
export * from './cell-image/cell-image.component';
export * from './outline-table/outline-table.component';
export * from './page-container/page-container.component';
export * from './page-content/page-content.component';
export * from './page-toolbar/page-toolbar.component';
export * from './table/table.component';
export * from './table-header/table-header.component';
export * from './table-row/table-row.component';
export * from './transaction-modal/transaction-modal.component';
export * from './confirm-transaction-modal/confirm-transaction-modal.component';
export * from './error-transaction-modal/error-transaction-modal.component';
export * from './drawer/drawer.component';
export * from './drawer-header/drawer-header.component';
export * from './drawer-item/drawer-item.component';
export * from './protected-route/protected-route.component';
export * from './float-button/float-button.component';
export * from './datetime/datetime.component';
export * from './app-toolbar/app-toolbar.component';
export * from './sales-report-modal/sales-report-modal.component';
export * from './toast/toast.component';
export * from './error-page/error-page.component';
export * from './progress/progress.component';
export * from './indeterminate-progress-bar/indeterminate-progress-bar.component';
export * from './alert/alert.component';
export * from './pagination/pagination.component';
export * from './text-item/text-item.component';
