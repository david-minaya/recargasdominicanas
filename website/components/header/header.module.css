.container {
  position: fixed;
  left: 0px;
  right: 0px;
  top: 0px;
  background-color: white;
  vertical-align: middle;
  z-index: 1;
}

.header {
  display: grid;
  grid-template-columns: auto auto;
  align-items: center;
  justify-items: start;
  padding: 12px 16px;
  margin: 0px auto;
  box-shadow: 0px 2px 6px lightgray;
}

.titleContainer {
  display: flex;
  flex-direction: row;
  align-items: center;
  text-decoration: none;
  -webkit-tap-highlight-color: transparent;
}

.icon {
  width: 28px;
  margin-right: 12px;
}

.title {
  color: var(--accent-color);
  font-size: 20px;
  font-weight: 500;
  white-space: nowrap;
}

.menuButton_base {
  width: 52px;
  justify-self: end;
  padding-top: 2px;
}

@media (min-width: 449px) {

  .header {
    padding: 12px 24px;
  }

  .icon {
    width: 32px;
    margin-right: 16px;
  }
  
  .title {
    width: auto;
    font-size: 24px;
  }

  .menuButton_base {
    padding-top: 4px;
  }
}
