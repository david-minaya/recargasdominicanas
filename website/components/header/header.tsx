import React from 'react';
import Link from 'next/link';
import { useMediaQuery } from '../../hooks';
import { Menu } from '../menu';
import { Button } from '../button';
import { mergeStyle, Style } from './header.module.css';

interface Props {
  style?: Style;
}

export function Header(props: Props) {

  const { style: customStyle } = props;

  const style = mergeStyle(customStyle);
  const isMobile = useMediaQuery('(max-width: 800px)');
  const [showMenu, setShowMenu] = React.useState(false);

  if (isMobile === undefined) return null;

  return (
    <div className={style.container}>
      <div className={style.header}>
        <Link href='/'>
          <a className={style.titleContainer} href='/'>
            <img className={style.icon} src='images/recargas_dominicanas_32x32.svg'/>
            <div className={style.title}>Recargas Dominicanas</div>
          </a>
        </Link>
        {isMobile &&
          <Button 
            style={style.menuButton} 
            text='Menu' 
            onClick={() => setShowMenu(!showMenu)} />
        }
        {!isMobile &&
          <Menu/>
        }
      </div>
      {isMobile &&
        <Menu show={showMenu}/>
      }
    </div>
  );
}
