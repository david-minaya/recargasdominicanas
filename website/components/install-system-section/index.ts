export * from './install-system-section';
