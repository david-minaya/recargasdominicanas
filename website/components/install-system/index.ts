export { InstallSystem } from './install-system';
