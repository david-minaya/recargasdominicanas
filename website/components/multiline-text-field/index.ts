export * from './multiline-text-field';
