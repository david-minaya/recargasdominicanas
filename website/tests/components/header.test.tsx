import React from 'react';
import { mount } from 'enzyme';
import { Header } from '../../components/header';

describe('render header component', () => {
  
  it('render mobile version', () => {
    mockMatchMedia(true);
    const wrapper = mount(<Header/>);
    const title = wrapper.find('.title');
    const menuButton = wrapper.find('.menuButton');
    expect(title.text()).toBe('Recargas Dominicanas');
    expect(menuButton.exists()).toBeTruthy();
  });
  
  it('open menu on the mobile version', () => {
    mockMatchMedia(true);
    const wrapper = mount(<Header/>);
    const menuButton = wrapper.find('.menuButton').first();
    menuButton.simulate('click');
    const menu = wrapper.find('.menu');
    expect(menu.exists()).toBeTruthy();
  });
  
  it('render desktop version', () => {
    mockMatchMedia(false);
    const wrapper = mount(<Header/>);
    const menu = wrapper.find('.menu');
    expect(menu.exists).toBeTruthy();
  });
});

describe('snapshot', () => {

  it('mobile version', () => {
    mockMatchMedia(true);
    const wrapper = mount(<Header/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('mobile version with the menu opened', () => {
    mockMatchMedia(true);
    const wrapper = mount(<Header/>);
    const menuButton = wrapper.find('.menuButton').first();
    menuButton.simulate('click');
    expect(wrapper).toMatchSnapshot();
  });

  it('desktop version', () => {
    mockMatchMedia(false);
    const wrapper = mount(<Header/>);
    expect(wrapper).toMatchSnapshot();
  });
});

function mockMatchMedia(isMatch: boolean) {
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(() => ({ 
      matches: isMatch,
      addListener: jest.fn() 
    })),
  });
}
