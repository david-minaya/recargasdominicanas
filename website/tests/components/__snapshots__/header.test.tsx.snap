// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot desktop version 1`] = `
<Header>
  <div
    className="header"
  >
    <Link
      href="/"
    >
      <a
        className="titleContainer"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <img
          className="icon"
          src="recargas_dominicanas_32x32.svg"
        />
        <div
          className="title"
        >
          Recargas Dominicanas
        </div>
      </a>
    </Link>
    <Menu>
      <div
        className="menu"
      >
        <Link
          href="/"
        >
          <a
            className="option"
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Preguntas frecuentes
          </a>
        </Link>
        <Button
          customStyle={
            Object {
              "default": "button",
            }
          }
          onClick={[Function]}
          text="Solicitar Sistema"
        >
          <button
            className="button"
            onClick={[Function]}
          >
            Solicitar Sistema
          </button>
        </Button>
      </div>
    </Menu>
  </div>
</Header>
`;

exports[`snapshot mobile version 1`] = `
<Header>
  <div
    className="header"
  >
    <Link
      href="/"
    >
      <a
        className="titleContainer"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <img
          className="icon"
          src="recargas_dominicanas_32x32.svg"
        />
        <div
          className="title"
        >
          Recargas Dominicanas
        </div>
      </a>
    </Link>
    <Button
      customStyle={
        Object {
          "default": "menuButton",
        }
      }
      onClick={[Function]}
      text="Menu"
    >
      <button
        className="menuButton"
        onClick={[Function]}
      >
        Menu
      </button>
    </Button>
  </div>
  <CSSTransition
    classNames={Object {}}
    in={false}
    timeout={
      Object {
        "enter": 900,
        "exit": 400,
      }
    }
    unmountOnExit={true}
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={false}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={
        Object {
          "enter": 900,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    />
  </CSSTransition>
</Header>
`;

exports[`snapshot mobile version with the menu opened 1`] = `
<Header>
  <div
    className="header"
  >
    <Link
      href="/"
    >
      <a
        className="titleContainer"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <img
          className="icon"
          src="recargas_dominicanas_32x32.svg"
        />
        <div
          className="title"
        >
          Recargas Dominicanas
        </div>
      </a>
    </Link>
    <Button
      customStyle={
        Object {
          "default": "menuButton",
        }
      }
      onClick={[Function]}
      text="Menu"
    >
      <button
        className="menuButton"
        onClick={[Function]}
      >
        Menu
      </button>
    </Button>
  </div>
  <CSSTransition
    classNames={Object {}}
    in={true}
    timeout={
      Object {
        "enter": 900,
        "exit": 400,
      }
    }
    unmountOnExit={true}
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={
        Object {
          "enter": 900,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <Menu>
        <div
          className="menu"
        >
          <Link
            href="/"
          >
            <a
              className="option"
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Preguntas frecuentes
            </a>
          </Link>
          <Button
            customStyle={
              Object {
                "default": "button",
              }
            }
            onClick={[Function]}
            text="Solicitar Sistema"
          >
            <button
              className="button"
              onClick={[Function]}
            >
              Solicitar Sistema
            </button>
          </Button>
        </div>
      </Menu>
    </Transition>
  </CSSTransition>
</Header>
`;
